src/main/java/com/pinterest/secor/io/impl/AvroFileReaderWriterFactory.java

```diff
@@ -117,23 +122,21 @@ public void close() throws IOException {
 
         private String topic;
 
-        public AvroFileWriter(LogFilePath logFilePath, CompressionCodec codec) throws IOException {
+        public AvroFileWriter(LogFilePath logFilePath, String avroCodec) throws IOException {
             file = new File(logFilePath.getLogFilePath());
             file.getParentFile().mkdirs();
             LOG.debug("Creating Brand new Writer for path {}", logFilePath.getLogFilePath());
             topic = logFilePath.getTopic();
             Schema schema = schemaRegistry.getSchema(topic);
             SpecificDatumWriter specificDatumWriter= new SpecificDatumWriter(schema);
             writer = new DataFileWriter(specificDatumWriter);
-            writer.setCodec(getCodecFactory(codec));
+            writer.setCodec(getCodecFactory(avroCodec));
             writer.create(schema, file);
         }
 
-        private CodecFactory getCodecFactory(CompressionCodec codec) {
-            CompressionCodecName codecName = CompressionCodecName
-                    .fromCompressionCodec(codec != null ? codec.getClass() : null);
+        private CodecFactory getCodecFactory(String avroCodec) {
             try {
-                return CodecFactory.fromString(codecName.name().toLowerCase());
+                return CodecFactory.fromString(avroCodec);
```

Comment:
Looks like you can have two edge cases: avroCode = "null" and avroCodec = null, maybe you need to deal with them separately

Reply:
Do you need to deal with avroCode == null || acvoCode.equalsIgnoreCase("none") here before calling CopdecFactory.fromSring?

Reply:
It's check in the CompressionUtil class